app/src/composables/tips.ts

```diff
@@ -0,0 +1,119 @@
+import { EthersJsAdapter } from '@/adapters/web3LibraryAdapter'
+import { TipsService } from '@/services/tipsService'
+import type { EventResult, TipsEventType } from '@/types'
+import dayjs from 'dayjs'
+import type { EventLog } from 'ethers'
+import { ref } from 'vue'
+
+const tipsService = new TipsService(EthersJsAdapter.getInstance())
+
+export function useTipsBalance() {
+  const balance = ref<string>('0')
+  const loading = ref(false)
+  const error = ref<any>(null)
+
+  async function getBalance() {
+    try {
+      loading.value = true
+      balance.value = await tipsService.getBalance()
+    } catch (err) {
+      error.value = err
+    } finally {
+      loading.value = false
+    }
+  }
+
+  return { balance, getBalance, loading, error }
+}
+
+export function usePushTip() {
+  const loading = ref(false)
+  const error = ref<any>(null)
+  const isSuccess = ref(false)
+
+  async function pushTip(addresses: string[], amount: number): Promise<void> {
+    try {
+      loading.value = true
+      await tipsService.pushTip(addresses, amount)
+      isSuccess.value = true
+    } catch (err) {
+      error.value = err
+    } finally {
+      loading.value = false
+    }
+  }
+
+  return { pushTip, loading, error, isSuccess }
+}
+
+export function useSendTip() {
+  const loading = ref(false)
+  const error = ref<any>(null)
+  const isSuccess = ref(false)
+
+  async function sendTip(addresses: string[], amount: number): Promise<void> {
+    try {
+      loading.value = true
+      await tipsService.sendTip(addresses, amount)
+      isSuccess.value = true
+    } catch (err) {
+      error.value = err
+    } finally {
+      loading.value = false
+    }
+  }
+
+  return { sendTip, loading, error, isSuccess }
```

Comment:
```suggestion
  const transaction= ref<any>(null) // Use the right type of transaction if exist
  const error = ref<any>(null)
  const isLoading = ref(false)
  const isSuccess = ref(false)

  async function sendTip(addresses: string[], amount: number): Promise<void> {
    try {
      loading.value = true
      await tipsService.sendTip(addresses, amount)
      isSuccess.value = true
    } catch (err) {
      error.value = err
    } finally {
      loading.value = false
    }
  }
  
   function execute(){
       transaction.value= await sendTip()
   }
  return { execute, transaction, IsLoading, error, isSuccess }
```

This is an example

Reply:
I use isLoading because you use IsSuccess, but for isSuccess i prefer isLoaded

Reply:
i still dont understand why we need to create seperate function between execute and sendTip. why dont we create execute that will do exactly what pushTip do?

Reply:
If you send tip in your component the result of send tip is not reactive. 

let say you have

```ts
 {sendTip, isloading}= useSendTips;
// then you will call the function send tips
const tx=await  sendTip()

Now you can't use the transaction as a reactiv state (because you have this value only when send tip is called)
```

The solution you have is to do 

```ts
const tx= ref(null)

tx.value=await sendTip()
```

As you can see it's more complicated

Reply:
but we can do this instead
```typescript
async function sendTip(addresses: string[], amount: number): Promise<void> {
    try {
      loading.value = true
      tx.value = await tipsService.sendTip(addresses, amount)
      isSuccess.value = true
    } catch (err) {
      error.value = err
    } finally {
      loading.value = false
    }
  }
  ```

Reply:
and for the isLoaded i think it's better to use it on `getEvents` for the `isSuccess` for sending transaction

Reply:
I don't understand